src/Coordinate.java

```diff
@@ -13,10 +13,15 @@ public Coordinate() {
 	// read about the observer pattern
 	//
 	public String call() throws Exception {
-		if (!this.called) {
+		if (!this.called && !this.hasShip) {
 			this.notifyObserver();
+			this.called = true;
 			return "Miss";
+		} else if (this.hasShip) {
+			this.observer.update();
+			return "Hit";
 		} else {
+			this.called = true;
 			throw new Exception("Coordinate already called");
```

Comment:
the above logic can be simplified. Call is doing too many things, breaking SOLID / clean code principles.